flake.nix

```diff
@@ -1,29 +1,134 @@
           pkgs = import nixpkgs {
             inherit system overlays;
           };
+          craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
+          crateNameFromCargoToml = craneLib.crateNameFromCargoToml {
+            cargoToml = ./packages/hoprd/crates/hoprd-hoprd/Cargo.toml;
+          };
+          solcDefault = with pkgs; (solc.mkDefault pkgs solc_0_8_19);
+          ethereumBindings = pkgs.stdenv.mkDerivation {
+            pname = "${crateNameFromCargoToml.pname}-ethereum-bindings";
+            version = crateNameFromCargoToml.version;
+            src = lib.fileset.toSource {
+              root = ./.;
+              fileset = fs.unions [
+                (fs.fileFilter (file: file.hasExt "sol") ./vendor/solidity)
+                (fs.fileFilter (file: file.hasExt "sol") ./packages/ethereum/contracts/src)
+                ./packages/ethereum/contracts/foundry.toml
+                ./packages/ethereum/contracts/remappings.txt
+              ];
+            };
+            buildInputs = with pkgs; [ foundry-bin solcDefault ];
+            buildPhase = ''
+              cd ./packages/ethereum/contracts
+              sed -i "s|solc = .*|solc = \"${solcDefault}/bin/solc\"|g" foundry.toml
+              forge bind --bindings-path ../crates/bindings --crate-name bindings \
+                --overwrite --select '^Hopr.*?(Boost|[^t])$$'
+            '';
+            installPhase = ''
+              echo `pwd`
+              cp -r --no-preserve=mode,ownership ../crates/bindings $out/
+            '';
+          };
+          src = fs.toSource {
+            root = ./.;
+            fileset = fs.unions [
+              ./vendor/cargo
+              ./.cargo/config.toml
+              ./Cargo.lock
+              (fs.fileFilter (file: file.hasExt "rs") ./.)
+              (fs.fileFilter (file: file.hasExt "toml") ./.)
+              ./packages/hoprd/crates/hopr-lib/data
+            ];
+          };
+          rustToolchain = pkgs.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+          nativeBuildInputs = with pkgs; [
+            rustToolchain
+            pkg-config
+          ];
+          buildInputs = with pkgs; [
+            foundry-bin
+          ] ++ pkgs.lib.optionals pkgs.stdenv.isDarwin (
+            with darwin.apple_sdk.frameworks; [
+              SystemConfiguration
+            ]
+          );
+          commonArgs = {
+            inherit (crateNameFromCargoToml) pname version;
+            inherit src buildInputs nativeBuildInputs;
+            CARGO_HOME = ".cargo";
+            cargoVendorDir = "./vendor/cargo";
+            # disable running tests automatically for now
+            doCheck = false;
+          };
+          cargoArtifacts = craneLib.buildDepsOnly (commonArgs // {
+            extraDummyScript = ''
+              rm -rf $out/vendor/cargo
+              cp -r --no-preserve=mode,ownership ${src}/vendor/cargo $out/vendor/
```

Comment:
What is the role of this script?

Reply:
I get it now, you sync all the vendored apps. Ok, so this is slow when done on deps, please, use `rsync`. 🍻